### PR TITLE
Introduce HMM_PowerF()

### PR DESCRIPTION
**Pull request by miblodelcarpio**
I'm probably missing something and, if so, I'll try and understand the problem with negative bases, but this function produces close to the same results (allowing for float precision, I believe) as WolframAlpha. [#29]

### CODE REVIEW

**Comment by bvisness** [comment]
This needs a header and a test before it can be merged.

**Comment by StrangeZak** [comment]
@miblodelcarpio Yeah just add a function declaration to the "header" part of the file then write a test case inside of hmm_test.cpp its pretty each you can just reference the other function to see how to do it. Also there is merge conflicts so if you can fix those that would be awesome

**Comment by bvisness** [comment]
Looking good! Try merging the latest `master` into your branch and resolving conflicts though - that will allow Travis to run unit tests for us.

**Comment by bvisness** [comment]
Whoa, not sure what's going on with Travis. Looking into that...

File: test/hmm_test.cpp [hunk 1]
```diff
@@ -64,6 +64,12 @@ TEST(ScalarMath, PowerF)
     EXPECT_FLOAT_EQ(HMM_PowerF(-2.0f, 4.1), -17.148376f);
     EXPECT_FLOAT_EQ(HMM_PowerF(2.0f, -2.5), 0.176777f);
     EXPECT_FLOAT_EQ(HMM_PowerF(-2.0f, -2.5), -0.176777f);
+
+TEST(ScalarMath, Lerp)
+{
+    EXPECT_FLOAT_EQ(HMM_Lerp(-2.0f, 0.0f, 2.0f), -2.0f);        
+    EXPECT_FLOAT_EQ(HMM_Lerp(-2.0f, 0.5f, 2.0f), 0.0f);        
+    EXPECT_FLOAT_EQ(HMM_Lerp(-2.0f, 1.0f, 2.0f), 2.0f);
```
Comment (bvisness) [inline test/hmm_test.cpp]: Looks like you may have inadvertently put a test inside another test.

Comment (bvisness) [inline test/hmm_test.cpp]: Or maybe it's just a missing closing brace. In any case, once that's fixed, Travis should be happy.

**Comment by miblodelcarpio** [comment]
Ah okay! Yeah, I thought I (and indeed) had fetched and merged the latest from this `master`. Travis must still be holding that grudge with me…

**Comment by miblodelcarpio** [comment]
Ach! Okay, fixing it now. Sorry, chaps!

**Comment by bvisness** [comment]
No worries! This is why we have the tests 😄

**Comment by miblodelcarpio** [comment]
Alright, I must have somehow deleted more than I should have when resolving the merge conflict. Retracing my steps to fix it all.

**Comment by bvisness** [comment]
Actually no, it's just that the results aren't exact enough for the tests to consider them equal. Try using `EXPECT_NEAR(actual, expected, absoluteError);` instead:

```c++
EXPECT_NEAR(HMM_PowerF(2.0f, -2.5), 0.176777f, 0.0001f);
```

**Comment by miblodelcarpio** [comment]
Huzzah! Thanks for your help with this, @bvisness and @StrangeZak (via IRC).

**Comment by bvisness** [comment]
Great! I will let @StrangeZak do the honors.

**Comment by StrangeZak** [comment]
@bvisness There's a bug in it i think give me a few.

**Comment by miblodelcarpio** [comment]
Sorry about that, chaps. With a bit of luck _this_ will do what it should do.

**Pull request merged**